presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -539,7 +552,14 @@ public ListeningExecutorService createResourceManagerExecutor(ResourceManagerCon
         configBinder(binder).bindConfig(StaticFunctionNamespaceStoreConfig.class);
         binder.bind(FunctionAndTypeManager.class).in(Scopes.SINGLETON);
         binder.bind(MetadataManager.class).in(Scopes.SINGLETON);
-        binder.bind(Metadata.class).to(MetadataManager.class).in(Scopes.SINGLETON);
+
+        if (serverConfig.isCatalogServerEnabled() && serverConfig.isCoordinator()) {
+            binder.bind(RemoteMetadataManager.class).in(Scopes.SINGLETON);
+            binder.bind(Metadata.class).to(RemoteMetadataManager.class).in(Scopes.SINGLETON);
+        }
+        else {
+            binder.bind(Metadata.class).to(MetadataManager.class).in(Scopes.SINGLETON);
+        }
```

Comment:
Can we put it in next commit or PR? This is a change to coordinator instead of catalog however this commit is "install catalog server"

Reply:
Ive squashed the commit so the message might apply to it now, if it still dosent make sense to have it on the main commit I can make another one for it